code/game/objects/items/storage/fancy.dm

```diff
@@ -99,7 +129,36 @@
 	for(var/i in 1 to 6)
 		new /obj/item/reagent_containers/food/snacks/donut/deadly(src)
 
+/obj/item/storage/fancy/donut_box/deadly/PopulateContents()
+	. = ..()
+	update_icon()
+
+/obj/item/storage/fancy/donut_box/deadly/update_icon_state()
+	. = ..()
+	if(fancy_open)
+		icon_state = "donutbox_inner"
+	else
+		icon_state = "donutbox"
+
+/obj/item/storage/fancy/donut_box/deadly/update_overlays()
+	. = ..()
+
+	if (!fancy_open)
+		return
+
+	var/donuts = 0
 
+	for (var/_donut in contents)
+		var/obj/item/reagent_containers/food/snacks/donut/donut = _donut
+		if (!istype(donut))
+			continue
+
+		. += image(icon = initial(icon), icon_state = donut.in_box_sprite(), pixel_x = donuts * DONUT_INBOX_SPRITE_WIDTH)
+		donuts += 1
+
+	. += image(icon = initial(icon), icon_state = "donutbox_top")
```

Comment:
This should all already exist from parent, no need to copy paste all of this

Reply:
Done but, sprite isn't showing correctly until you remove a donut from the box, not sure how to fix that.